pkg/controller/operators/catalog/operator.go

```diff
@@ -1033,6 +1042,84 @@ func (o *Operator) createInstallPlan(namespace string, subs []*v1alpha1.Subscrip
 	return reference.GetReference(res)
 }
 
+func (o *Operator) checkBundleLookups(plan *v1alpha1.InstallPlan) (bool, error) {
+	for _, bundleLookup := range plan.Status.BundleLookups {
+		if bundleLookup.BundleJob == nil {
+			configmap, job, err := configmap.LaunchBundleImage(o.opClient.KubernetesInterface(), bundleLookup.Image, "quay.io/jpeeler/bundle-init-image:latest", o.namespace)
+			if err != nil {
+				return false, err
+			}
+			logrus.Infof("Launched bundle job for image %v", bundleLookup.Image)
+
+			bundleLookup.BundleJob = &v1alpha1.BundleJob{
+				// job condition and completion time will be filled out later (installplan sync)
+				Name:      job.GetName(),
+				Namespace: job.GetNamespace(),
+			}
+			bundleLookup.ConfigMapRef = &v1alpha1.ConfigMapResourceReference{
+				Name:            configmap.GetName(),
+				Namespace:       configmap.GetNamespace(),
+				UID:             configmap.GetUID(),
+				ResourceVersion: configmap.GetResourceVersion(),
+			}
+			_, err = o.client.OperatorsV1alpha1().InstallPlans(plan.GetNamespace()).UpdateStatus(plan)
+			if err != nil {
+				return false, err
+			}
+
+			return false, nil
+		}
+
+		if bundleLookup.BundleJob.CompletionTime != nil {
+			// already processed
+			return true, nil
+		}
+
+		// TODO: instead of doing a get, should just watch all jobs (add ownerref) and update the installplan from there
+		job, err := o.opClient.KubernetesInterface().BatchV1().Jobs(bundleLookup.BundleJob.Namespace).Get(bundleLookup.BundleJob.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if len(job.Status.Conditions) == 0 || len(job.Status.Conditions) > 0 && job.Status.Conditions[0].Type != batchv1.JobComplete {
+			logrus.Infof("Job '%v' for '%v' not yet completed", job.GetName(), bundleLookup.Image)
+			return false, nil
+		}
+		bundleLookup.BundleJob.Condition = job.Status.Conditions[0].Type
+		bundleLookup.BundleJob.CompletionTime = job.Status.CompletionTime
+
+		configmap, err := o.opClient.KubernetesInterface().CoreV1().ConfigMaps(bundleLookup.ConfigMapRef.Namespace).Get(bundleLookup.ConfigMapRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		// extract data from configmap and write to install plan
+		bundle, err := o.bundleLoader.Load(configmap)
```

Comment:
That makes sense but is not critical, IMO. Installing via installplan steps ensures we install via a well-tested path. But, this would be a great followup and will enable installplans to install larger sets of objects.